File: programmers/동영상_재생기.py
def solution(video_len, pos, op_start, op_end, commands):
    video_seconds = time_string_to_seconds_int(video_len)
    pos_seconds = time_string_to_seconds_int(pos)
    opening_start_seconds = time_string_to_seconds_int(op_start)
    opening_end_seconds = time_string_to_seconds_int(op_end)

    pos_seconds = opening_skip(pos_seconds, opening_start_seconds, opening_end_seconds)

    for command in commands:
        if command == "next":
            pos_seconds += 10
        elif command == "prev":
            pos_seconds -= 10
        if pos_seconds < 0:
            pos_seconds = 0
        pos_seconds = opening_skip(
            pos_seconds, opening_start_seconds, opening_end_seconds
        )
        if pos_seconds > video_seconds:
            pos_seconds = video_seconds

    return f"{string_time_fomater(pos_seconds // 60)}:{string_time_fomater(pos_seconds % 60)}"


def time_string_to_seconds_int(time_str):
    time_m, time_s = map(int, time_str.split(":"))
    return time_m * 60 + time_s


def string_time_fomater(num):
    return f"0{num}" if num < 10 else f"{num}"


def opening_skip(s, op_start, op_end):
    if op_start <= s < op_end:
        return op_end
    else:
        return s

File: programmers/동영상_재생기.java
class Solution {
    private int videoSec;
    private int presentSec;
    private int opStartSec;
    private int opEndSec;

    public String solution(String video_len, String pos, String op_start, String op_end, String[] commands) {
        this.videoSec = getSec(video_len);
        this.presentSec = getSec(pos);
        this.opStartSec = getSec(op_start);
        this.opEndSec = getSec(op_end);

        jumpOp();

        for (String command : commands) {
            commandControl(command);
        }

        return toStringFormat();
    }
    
    private void commandControl(String command) {
        if (command.equals("next")) {
            presentSec += 10;
            if (this.presentSec > this.videoSec)
                this.presentSec = this.videoSec;
        } else {
            this.presentSec -= 10;
            if (this.presentSec < 0)
                this.presentSec = 0;
        }
        jumpOp();
    }
    
    private int getSec(String strTime) {
        String[] strSplit = strTime.split(":");
        return Integer.parseInt(strSplit[0]) * 60 + Integer.parseInt(strSplit[1]);
    }
    
    private String toStringFormat() {
        String minStr = String.valueOf(this.presentSec / 60);
        String secStr = String.valueOf(this.presentSec % 60);

        if (minStr.length() == 1)
            minStr = "0" + minStr;
        if (secStr.length() == 1)
            secStr = "0" + secStr;

        return minStr + ":" + secStr;
    }

    private void jumpOp() {
        if (this.opStartSec <= this.presentSec & this.presentSec <= this.opEndSec) {
            this.presentSec = this.opEndSec;
        }
    }
}